src/core/stdc/assert_.d

```diff
@@ -0,0 +1,52 @@
+/**
+ * D header file for C99.
+ *
+ * $(C_HEADER_DESCRIPTION pubs.opengroup.org/onlinepubs/009695399/basedefs/_assert.h.html, _assert.h)
+ *
+ * License: Distributed under the
+ *      $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0).
+ *    (See accompanying file LICENSE)
+ * Source:    $(DRUNTIMESRC core/stdc/_assert_.d)
+ * Standards: ISO/IEC 9899:1999 (E)
+ */
+
+module core.stdc.assert_;
+
+extern (C):
+@trusted:
+nothrow:
+@nogc:
```

Comment:
As far as I understand, this provides an alternative to built-in `assert`. Since the built-in `assert` is usable from `pure` functions, I think this should also be.

> I'm not sure that a function never returning means it is pure.

That should never happen in correct programs. But in general `pure` functions are allowed to terminate the program, [according to the language spec](http://dlang.org/spec/function.html#pure-functions). So I think the C `assert` functions fit the bill.